### DISCUSSION TITLE
About performance

### DISCUSSION THREAD

**Discussion by vmsh0** [any1/wayvnc#287] [Q&A]
Hello,

I'm attempting to create a "virtual docking mode" for my laptop, to use it remotely from my desktop PC when I'm physically in the office. I'm running the latest stable Sway on Arch Linux, and wayvnc works nicely to transmit the display to my desktop PC. The desktop PC runs Windows 11 and TightVNC Viewer.

What I'm experiencing is a system that definitely works, but with poor performance. When watching a YouTube video (I'm using [a random dashcam video](https://www.youtube.com/watch?v=4e4iE1FxRkk) as a reference to have a continously-moving image on the screen) I can visibly see individual blocks on the screen updating in the usual top-bottom scan patten, instead of a smooth image. And this is with a relatively low image quality, too.

Network traffic during this test peaks at about ~12Mbps, while `iperf3` shows that there is a stable >50Mbps bandwidth available between the two machines, which are on the same Wi-Fi network. I've tried using the `-g` flag, I haven't noticed anything different. (I have both a discrete AMD GPU and an integrated Intel GPU in this 10th gen Intel Core laptop)

Before going further down the road, I wanted to assess what are the correct expectations for this kind of setup:

- How can I find out what image encoding am I using now? Visually, it looks a lot like 80-ish quality JPEG, can I expect better results with a different encoding? Is that supported by `wayvnc`? (I've read the manual page and the FAQ, and I've also fumbled with the available command line flags, but I haven't found this information)
- I'm not interested in a perfectly-seamless setup (e.g. remote gaming type of quality), but as an example, it would be nice to be able to hold a video-call with reasonable fluidity. Am I perhaps just using the wrong tool for the job? What would be a more suitable alternative?

**Comment by any1** [comment]
I recommend that you use the TigerVNC client rather than TightVNC. It supports h264 encoding.

You need to run wayvnc with the `--gpu` flag for h264 to be enabled.

If you enable verbose logging in wayvnc, you will see which encoding is chosen for a newly connected client. You can choose encoding in TigerVNC if you click "options" and select the "Compression" tab. Select H.264 for the best result. To adjust quality you can set the jpeg compression quality. Second best option is Tight encoding with lossy/jpeg encoding allowed. If you want perfect picture, disable jpeg.

The capture rate is capped at 30 fps by default; you can adjust that limit using the `--max-fps` flag. If you want 60 fps, I recommend setting it to 120 so that it definitely does not interfere.

**Reply by vmsh0** [reply]
I have tried using H.264, but wayvnc prints an endless stream of complaints:

```
ERROR: libav: 0: Failed to initialise VAAPI connection: -1 (unknown libva error).
ERROR: libav: 0: Failed to created derived device context: -5.
ERROR: libav: 0: Failed to configure output pad on Parsed_hwmap_0
ERROR: ../neatvnc/src/server.c: 1151: Failed to encode current frame
```

This is on:
```
~
➜ wayvnc -V
wayvnc: v0.7.2-0-g4de85ac (makepkg)
neatvnc: v0.7.1-0-gb5b330b (makepkg)
aml: v0.3.0-0-gb83f357 (makepkg)
```

**Reply by vmsh0** [reply]
Hey,
For reference, this happens on a second laptop I tested (identical system, moved the hard drive).
